hibernate-core/src/main/java/org/hibernate/generator/internal/CurrentTimestampGeneration.java

```diff
@@ -255,6 +262,11 @@ public EnumSet<EventType> getEventTypes() {
 		return eventTypes;
 	}
 
+	@Override
+	public boolean allowMutation() {
+		return allowMutation;
+	}
+
```

Comment:
@yrodiere This is the only significant thing here.

Reply:
> LGTM, thanks, but should we merge this immediately, or do you want someone to have a look at the "manually set the value for inserts" case too?

Well, nobody actually asked for that yet. (Except you 😉)

> Asking because that might change the exposed API, if we decide its' better (simpler) to expose a single `allowOverride()` attribute controlling whether people can manually set the value for both inserts and updates.

It's not linked directly to inserts vs updates, it depends on whether the client can override generation. So I think it's two different things.

All that said, we can't merge this without a test.